File: app/src/main/java/com/msk/produtosperigosos/listas/ListaInfracoes.java
package com.msk.produtosperigosos.listas;

import android.content.res.Resources;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import androidx.appcompat.widget.AppCompatEditText;
import androidx.fragment.app.ListFragment;

import com.msk.produtosperigosos.R;
import com.msk.produtosperigosos.info.Infracoes;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class ListaInfracoes extends ListFragment implements TextWatcher {

    SimpleAdapter adapter;
    List<HashMap<String, String>> listaCompleta;
    HashMap<String, String> hm;
    Resources r = null;
    private AppCompatEditText textoPesquisado;
    private String[] itemlinha;
    private int[] linhaLista;
    private View tela;

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        textoPesquisado = tela.findViewById(R.id.etNomePesquisado);
        listaCompleta = new ArrayList<HashMap<String, String>>();

        r = getResources();

        for (int i = 0; i < Infracoes.codigoInfacao.length; i++) {
            hm = new HashMap<String, String>();
            hm.put("codigo", Infracoes.codigoInfacao[i]);
            hm.put("amparo", Infracoes.amparoInfacao[i]);
            hm.put("descricao", Infracoes.descricaoInfacao[i]);
            listaCompleta.add(hm);
        }

        itemlinha = new String[]{"codigo", "amparo", "descricao"};

        linhaLista = new int[]{R.id.tvCodigo, R.id.tvAmparo, R.id.tvInfracao};

        adapter = new SimpleAdapter(getActivity(), listaCompleta, R.layout.item_infracao, itemlinha, linhaLista);

        setListAdapter(adapter);

        textoPesquisado.addTextChangedListener(this);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        tela = inflater.inflate(R.layout.lista_infracoes, container, false);

        return tela;
    }

    @Override
    public void onListItemClick(ListView l, View v, int posicao, long id) {

        // ACAO QUANDO ALGUM ITEM FOR CLICADO

    }

    @Override
    public void afterTextChanged(Editable arg0) {

    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count,
                                  int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {

        ListaInfracoes.this.adapter.getFilter().filter(s);
    }


}

File: src/main/java/com/msk/produtosperigosos/listas/ListaInfracoes.kt
package com.msk.produtosperigosos.listas

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListView
import android.widget.SimpleAdapter
import androidx.appcompat.widget.AppCompatEditText
import androidx.fragment.app.ListFragment
import com.msk.produtosperigosos.R
import com.msk.produtosperigosos.info.Infracoes

class ListaInfracoes : ListFragment(), TextWatcher {

    private lateinit var adapter: SimpleAdapter
    private lateinit var textoPesquisado: AppCompatEditText

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.lista_infracoes, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        textoPesquisado = view.findViewById(R.id.etNomePesquisado)

        val listaCompleta = ArrayList<HashMap<String, String>>()

        for (i in Infracoes.codigoInfracao.indices) {
            val hm = HashMap<String, String>()
            hm["codigo"] = Infracoes.codigoInfracao[i]
            hm["amparo"] = Infracoes.amparoInfacao[i]
            hm["descricao"] = Infracoes.descricaoInfacao[i]
            listaCompleta.add(hm)
        }

        val itemlinha = arrayOf("codigo", "amparo", "descricao")

        val linhaLista = intArrayOf(R.id.tvCodigo, R.id.tvAmparo, R.id.tvInfracao)

        adapter = SimpleAdapter(
            requireActivity(),
            listaCompleta,
            R.layout.item_infracao,
            itemlinha,
            linhaLista
        )

        listAdapter = adapter

        textoPesquisado.addTextChangedListener(this)
    }

    override fun onListItemClick(l: ListView, v: View, position: Int, id: Long) {
        // ACAO QUANDO ALGUM ITEM FOR CLICADO
    }

    override fun afterTextChanged(arg0: Editable?) {
    }

    override fun beforeTextChanged(
        s: CharSequence?, start: Int, count: Int,
        after: Int
    ) {
    }

    override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
        adapter.filter.filter(s)
    }
}
